src/VisualStudio/Core/Def/Implementation/Preview/TopLevelChange.cs

```diff
@@ -95,15 +95,15 @@ private Solution ApplyFileChanges(Solution solution, IEnumerable<FileChange> fil
                         var newLinkedDocumentIdOpt = updatedDocumentTextOpt != null ? oldLinkedDocument.Id : null;
 
                         ApplyFileChangesCore(oldLinkedDocument, newLinkedDocumentIdOpt, updatedDocumentTextOpt,
-                            fileChange.CheckState, fileChange.IsAdditionalDocumentChange);
+                            fileChange.CheckState, fileChange.IsAdditionalDocumentChange, fileChange.IsAnalyzerConfigDocumentChange);
                     }
                 }
                 else if (updatedTextDocument is Document updatedDocument)
                 {
                     foreach (var newLinkedDocumentId in updatedDocument.GetLinkedDocumentIds())
                     {
                         ApplyFileChangesCore(oldTextDocument, newLinkedDocumentId, updatedDocumentTextOpt,
-                            fileChange.CheckState, fileChange.IsAdditionalDocumentChange);
+                            fileChange.CheckState, fileChange.IsAdditionalDocumentChange, fileChange.IsAnalyzerConfigDocumentChange);
```

Comment:
I feel that there should be an enum stating the type of the doc (Normal/Additional/Analyzer), and we can pass that along.  code that processed can operate on that enum and throw when they encounter a new type.  that will make our code at least *throwing* when we add new types, instead of silently doing the wrong thing.